test/distributed/org/apache/cassandra/distributed/mock/nodetool/NetstatsTest.java

```diff
@@ -0,0 +1,607 @@
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.datastax.driver.core.Session;
+import org.apache.cassandra.distributed.Cluster;
+import org.apache.cassandra.distributed.api.IInvokableInstance;
+import org.apache.cassandra.distributed.api.IIsolatedExecutor;
+import org.apache.cassandra.distributed.impl.InstanceAwareNodetool;
+import org.apache.cassandra.distributed.shared.AbstractBuilder;
+import org.apache.cassandra.distributed.shared.NetworkTopology;
+import org.apache.cassandra.distributed.test.TestBaseImpl;
+import org.apache.cassandra.service.StorageService;
+import org.apache.cassandra.utils.Pair;
+
+import static java.util.stream.Collectors.toList;
+import static org.apache.cassandra.distributed.api.Feature.GOSSIP;
+import static org.apache.cassandra.distributed.api.Feature.NATIVE_PROTOCOL;
+import static org.apache.cassandra.distributed.api.Feature.NETWORK;
+
+public class NetstatsTest extends TestBaseImpl
+{
+
+    private static final Logger logger = LoggerFactory.getLogger(NetstatsTest.class);
+
+    @Test
+    public void testNetstatsStreamingProgress() throws Exception
+    {
+        final ExecutorService executorService = Executors.newFixedThreadPool(2);
+
+        try (final InstanceAwareNodetool nodetool = new InstanceAwareNodetool(getCluster()))
+        {
+            final Cluster cluster = nodetool.getCluster();
+            final IInvokableInstance node1 = cluster.get(1);
+            final IInvokableInstance node2 = cluster.get(2);
+
+            createTable(cluster);
+
+            node1.flush("netstats_test");
+            node2.flush("netstats_test");
+
+            disableCompaction(node1);
+            disableCompaction(node2);
+
+            populateData();
+
+            // change RF from 1 to 2 so we need to repair it, repairing will causes streaming shown in netstats
+            changeReplicationFactor();
+
+            final Future<NetstatResults> resultsFuture1 = executorService.submit(new NetstatsCallable(nodetool, node1));
+            final Future<NetstatResults> resultsFuture2 = executorService.submit(new NetstatsCallable(nodetool, node2));
+
+            final IIsolatedExecutor.CallableNoExcept<?> repairCallable = node1.asyncCallsOnInstance(() -> {
+                try
+                {
+                    StorageService.instance.repair("netstats_test", Collections.emptyMap(), Collections.emptyList()).right.get();
+
+                    return null;
+                }
+                catch (final Exception ex)
+                {
+                    throw new RuntimeException(ex);
+                }
+            });
+
+            repairCallable.call();
```

Comment:
can simplify and just do

```
node1.asyncCallsOnInstance(() -> { ... }).call();
```

Looks like the API is missing asyncCallOnInstance which would do that for you... =(